.vscode/settings.json

```diff
@@ -29,6 +29,7 @@
     }
   },
   "eslint.rules.customizations": [
-    { "rule": "prettier/prettier", "severity": "off" }
+    { "rule": "prettier/prettier", "severity": "off" },
+    { "rule": "import/no-unresolved", "severity": "off" }
```

Comment:
This is because TypeScript declaration files are a strange thing, part source (when they're being imported by other packages) and part build artifact (when they're sitting quietly in the `dist` directory).

The TypeScript language server is smart enough to build its own model of the types, and this is how VSCode is able to do code navigation. But `eslint` isn't smart enough to do this, and will flag missing type declaration files as a lint error. This means that the codebase fails linting on a fresh checkout, which seems wrong (and is unfixable unless we check the type declarations in, which also seems wrong as they are build artifacts).

This configuration change only affects whether VSCode will display the error, not whether `eslint` will detect it, so it still gets picked up when running `yarn lint`. In CI, `yarn lint` runs after `yarn build`, which means that the type declaration files will be present.

Finally, if you're actually trying to import a package that doesn't exist, or which doesn't have an `index.ts`, or is not correctly configured to export types, then you will get a different (and better) error from the TypeScript language server in VSCode anyway, making this one redundant.

Reply:
makes sense!